### ISSUE TITLE
Feature request: pressing up arrow key on smart chat retrieve the last input

### ISSUE THREAD

**Issue by oscaromsn** [brianpetro/obsidian-smart-connections#499]
Hey there!

Firstly, I want to thank you again all the work on this plugin. I'm having a really nice experience with it and am keeping up with the latest updates and enjoying them.

As the title of this request suggests, is to enable the smart chat to function like a terminal, retrieving the inputs previously entered when pressing the up arrow key.

This feature is useful because it makes it easier to repeat prompts which involves file references. Therefore, it allows new and more complex prompt approaches, such as using static checklists on mutable files or chaining prompts (e.g., "[[context]] [[instructions check-list]] [[examples]] [[mutable file # 1]] [[final considerations]] — where sometimes I have the same context but an different task, or an example list which is constantly changing, etc."). 

Overall, terminal-like behavior promotes the use of LLMs more as reasoning assistants than just information retrievers — which I personally think is a clever AI use case in a note-taking app like Obsidian. 

At worst, the unadvised user might not even notice the difference.

Just to touch on the subject, I believe that features requested in other threads, such as autocomplete, would be more useful in the prompt box than in documents (and more cost-effective in this case).

Excited to hear other opinions and eventually see implemented :)

**Comment by bbecausereasonss** [comment]
This would save me SOO much time going back to old notes and copy/pasting. :)

**Comment by brianpetro** [comment]
Makes sense. I use this feature all the time in the terminal/CLI and probably even rely on it too much 🤣

🌴
